datashader/glyphs/line.py

```diff
@@ -547,19 +561,131 @@ def map_onto_pixel(sx, tx, sy, ty, xmin, xmax, ymin, ymax, x, y):
 
     return map_onto_pixel
 
+@ngjit
+def _xiaolinwu(x0, x1, y0, y1, agg):
+    """ Implementation of Xiaolin Wu's anti-aliasing algorithm for lines.
+    Loosely based on: https://rosettacode.org/wiki/Xiaolin_Wu%27s_line_algorithm#Python
+    """
+    dx, dy = x1-x0, y1-y0
+    steep = abs(dx) < abs(dy)
+    if steep:
+        x0, y0, x1, y1, dx, dy = y0, x0, y1, x1, dy, dx
+    if x1 < x0:
+        x0, x1, y0, y1 = x1, x0, y1, y0
+
+    grad = dy/dx
+    intery = y0 + _myrfpart(x0) * grad
+    xstart = _draw_endpoint(agg, steep, (x0, y0), grad) + 1
+    xend = _draw_endpoint(agg, steep,   (x1, y1), grad)
+    for x in range(xstart, xend):
+        y = int(intery)
+        _draw_pixel(agg, _flipxy(x, y, steep), _myrfpart(intery))
+        _draw_pixel(agg, _flipxy(x, y+1, steep), _myfpart(intery))
+        intery += grad
+
+@ngjit
+def _draw_pixel(agg, px, value):
+    """ Xiaolin Wu utility. """
```

Comment:
```suggestion
    """ Xiaolin Wu utility. """
```

It's great for methods to have docstrings, but only if the docstring conveys something beyond what's already clear from the method and class names.

Reply:
Yes, they clearly can't stay like this. I could either remove them or flesh them out a bit more. What is your preference?

Reply:
More info would be great.

Reply:
I will resolve this now as the code in question no longer exists.